cartoframes/viz/style.py

```diff
@@ -60,33 +60,36 @@ def _init_style(self, data):
         else:
             raise ValueError('`style` must be a string or a dictionary')
 
-    def compute_viz(self, geom_type, variables={}):
+    def compute_viz(self, geom_type, variables={}, properties={}):
         style = self._style
         default_style = defaults.STYLE[geom_type]
+
         if isinstance(style, dict):
             if geom_type in style:
                 style = style.get(geom_type)
-            return self._parse_style_dict(style, default_style, variables)
+            return self._parse_style_dict(style, default_style, variables, properties)
         elif isinstance(style, str):
-            return self._parse_style_str(style, default_style, variables)
+            return self._parse_style_str(style, default_style, variables, properties)
         else:
             raise ValueError('`style` must be a string or a dictionary')
 
-    def _parse_style_dict(self, style, default_style, ext_vars):
+    def _parse_style_dict(self, style, default_style, ext_vars, ext_props):
         variables = merge_dicts(style.get('vars', {}), ext_vars)
         properties = merge_dicts(default_style, style)
+        allproperties = merge_dicts(properties, ext_props)
 
         serialized_variables = self._serialize_variables(variables)
-        serialized_properties = self._serialize_properties(properties)
+        serialized_properties = self._serialize_properties(allproperties)
 
         return serialized_variables + serialized_properties
 
-    def _parse_style_str(self, style, default_style, ext_vars):
+    def _parse_style_str(self, style, default_style, ext_vars, ext_props):
         variables = ext_vars
         default_properties = self._prune_defaults(default_style, style)
+        properties = merge_dicts(default_properties, ext_props)
```

Comment:
New tests are required here too.

NOTE: `prune_defaults` removes the default properties from the string in case it is in the style. For example, if the style contains `color` this is removed from the default styles to avoid having `color` twice in the string. But this check is not made with the ext_props, so for example, if the user has a filter in the style, and ext_props contains a filter this will generate an error.